### PR TITLE
Script Updated

### PR DESCRIPTION
**Pull request by parveen9719**

### CODE REVIEW

File: cypress/e2e/pages/EmployeelistPage.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+get department() { return cy.get("#department")}
+get totalCount() { return cy.get("div[class='total'] span")}
+
+//Methods
+enterNameToSearch(nametxt) {
+  this.searchByName.clear().type(nametxt).should('have.value', nametxt);
+  cy.log("Entered Name to Search")
+  }  
+
+validateNoRecordsAppear(){
```
Comment (parveen9719) [inline cypress/e2e/pages/EmployeelistPage.js:20]: move in test class

File: cypress/e2e/pages/EmployeelistPage.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  this.noRecordAvailable.should('have.text', "No Records Available");
+  cy.log("No Records Appear")
+  }    
+
+countTotalEmployees(countEmployees){
```
Comment (parveen9719) [inline cypress/e2e/pages/EmployeelistPage.js:25]: this method should return count of employee and then we will assert it in test

File: cypress/e2e/pages/EmployeelistPage.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  this.employeeCount.invoke('text')
+  .then((text) => {
+  const employeeCount = parseInt(text.replace('Total Employees : ', '').trim());
+  expect(employeeCount).to.equal(countEmployees);  
+  cy.log(`Total number of employees: ${employeeCount}`);
+        });
+      }
+
+selectUser(){
```
Comment (parveen9719) [inline cypress/e2e/pages/EmployeelistPage.js:34]: this method should accept name of user

File: cypress/e2e/pages/HRApprovalPage.js [hunk 4]
```diff
@@ -15,22 +15,23 @@ get approveButton() { return cy.get("div[role='tabpanel'] form div button[type='
 
 // Methods
 selectDepartment(departmentName) {
-    this.department.type(departmentName + '{enter}').should('contain', departmentName);
+    //this.department.type(departmentName + '{enter}').should('contain', departmentName);
```
Comment (parveen9719) [inline cypress/e2e/pages/HRApprovalPage.js:18]: remove this

File: cypress/e2e/pages/UserDashboardPage.js [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
   // Locators    
   get workExperience() { return cy.get("h2[id='heading4'] button[type='button']") }
   get noRecordAvailable() { return cy.get("#collapse4 div.table-responsive div") }
   get basicInfo() { return cy.get("h2[id='heading1'] button[type='button']") }
   get editBasicInfo() { return cy.get("#collapse1 a i") }
   get closeButton() { return cy.get("button[class='btn btn-secondary btn-sm ms-2']") }
-  get editFirstName() { return cy.get("input[value='Auto']") }
-  get editLastName() { return cy.get("input[value='User']") }
-  get editMiddleName() { return cy.get("input[value='Mation']") }
+  get editFirstName() { return cy.get("input[name='firstName']:not(:disabled)") }
```
Comment (parveen9719) [inline cypress/e2e/pages/UserDashboardPage.js:12]: edit from name can be remove

File: cypress/e2e/tests/EmployeeAssetMgmtTests.cy.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        cy.login();
+
+        // Navigate To employee list Page 
+        sideBar.navigateTo("Employee Management", "Employees List");
+        
+        // Validate employee list page
+        EmployeelistPage.employeelistHeader.should('be.visible').and('have.text', "Employees List");
+        
+        // enter employee name 
+        EmployeelistPage.enterEmployeeName("DDinesh D Kumar");
```
Comment (parveen9719) [inline cypress/e2e/tests/EmployeeAssetMgmtTests.cy.js:20]: use automation user

File: cypress/e2e/tests/EmployeeAssetMgmtTests.cy.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import sideBar from "../components/SideBar";
```
Comment (parveen9719) [inline cypress/e2e/tests/EmployeeAssetMgmtTests.cy.js:1]: remove all extra spaces

File: cypress/fixtures/data.json [hunk 8]
```diff
@@ -1,3 +1,6 @@
 {
```
Comment (parveen9719) [inline cypress/fixtures/data.json:1]: remove unused  data

File: cypress/e2e/tests/EmployeeOnboardingTests.cy.js [hunk 9]
```diff
@@ -21,11 +21,16 @@ describe("Employee Onboard Tests", () => {
         const joineePersonalMail = generateRandomYopmail(10);
         invitations.enterEmailID(joineePersonalMail);
         invitations.enterEmployeeName("Mattews");
+
+        // Why should we need to pass the full path of the file ,
```
Comment (parveen9719) [inline cypress/e2e/tests/EmployeeOnboardingTests.cy.js:25]: remove comments

File: cypress/e2e/tests/EmployeeOnboardingTests.cy.js [hunk 10]
```diff
@@ -21,11 +21,16 @@ describe("Employee Onboard Tests", () => {
...
+        // we can create a comman path for fixture and append only different files
         invitations.selectSamplePdf('cypress/fixtures/resources/dummy.pdf');
+
         invitations.clickSubmitButton();
 
         // Verify mail sent notification is displayed
-        invitations.validateOnboardingEmailSentMsg('Onboarding welcome mail sent');
+        invitations.onBoardingSuccessMsg.should('contain.text', 'Onboarding welcome mail sent');
+        //invitations.validateOnboardingEmailSentMsg('Onboarding welcome mail sent');
```
Comment (parveen9719) [inline cypress/e2e/tests/EmployeeOnboardingTests.cy.js:33]: remove

File: cypress/e2e/pages/EmployeelistPage.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+get totalCount() { return cy.get("div[class='total'] span")}
+
+//Methods
+enterNameToSearch(nametxt) {
+  this.searchByName.clear().type(nametxt).should('have.value', nametxt);
+  cy.log("Entered Name to Search")
+  }  
+   
+
+countTotalEmployees() {
```
Comment (Sandy-N) [inline cypress/e2e/pages/EmployeelistPage.js:21]: getTotalEmployeescount

File: cypress/e2e/tests/ManagerAssigneeMgmtTests.cy.js [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+
+describe("Employee Assign Manager Managment Tests", () => {
+
+    it("HRMIS_1: Verify Assignee Manager Open when user click on Assing Manager Subtab", () => {
+
+        const expectedTexts = ['Id', 'Name', 'Assigned Manager', 'Action'];
+
+
+         AssignManagerPage.assignManagerLbl.invoke('text').then((text)=>{
```
Comment (Sandy-N) [inline cypress/e2e/tests/ManagerAssigneeMgmtTests.cy.js:23]: fix this page

File: cypress/e2e/tests/PromoteEmployeeTests.cy.js [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+    it("HRMIS_1: Verify Promote Open when user click on Promote Employee Subtab under Employee Managment", () => {
+
+        const expectedTexts = ['Employee Id', 'Name', 'Department', 'Designation' ,'Action'];
+
+        PromoteEmployeePage.promoteEmployeeLbl.invoke('text').then((text)=>{
+            expect(text.trim()).to.eq('Promote Employee');
+        });
+        cy.wait(1000);
+        PromoteEmployeePage.selectEmployee("Auto Mation User");
+        PromoteEmployeePage.tableHeadersLbl.each((headerText, index) => {
```
Comment (Sandy-N) [inline cypress/e2e/tests/PromoteEmployeeTests.cy.js:26]: fix

File: cypress/e2e/tests/PromoteEmployeeTests.cy.js [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+    it("HRMIS_3: Verify Validation message appears after clicking on submit button when no department or designation is selected", () => {
+        
+        cy.wait(1000);
+        PromoteEmployeePage.selectEmployee("Auto Mation User");
+        PromoteEmployeePage.clickOnPromote();
+        PromoteEmployeePage.promotePopHeaderLbl.should('have.text','Promote Employee');
+
+        // When no department is selected.
+        PromoteEmployeePage.clickOnSubmitBtn();
+        PromoteEmployeePage.selectDepartmentDrp
```
Comment (Sandy-N) [inline cypress/e2e/tests/PromoteEmployeeTests.cy.js:59]: fix

File: cypress/e2e/tests/PromoteEmployeeTests.cy.js [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+        
+
+        const designations = ['Select Designation','CEO & Chief Architect','Co Founder & COO','VP Sucess Manager & Sales','Chief People Officer','Piyush Dogra'];
+
+        cy.wait(1000);
+        PromoteEmployeePage.selectEmployee("Auto Mation User");
+        PromoteEmployeePage.clickOnPromote();
+        PromoteEmployeePage.promotePopHeaderLbl.should('have.text','Promote Employee');
+        PromoteEmployeePage.selectDepartment("GENRIC")
+        PromoteEmployeePage.getdesgnationOptions().then((options) => {
```
Comment (Sandy-N) [inline cypress/e2e/tests/PromoteEmployeeTests.cy.js:86]: fix

File: cypress/support/commands.js [hunk 16]
```diff
@@ -44,3 +44,51 @@ Cypress.Commands.add('logout', () => {
...
+    cy.wrap(element).trigger('keydown', { key: 'Enter' })
+);
+
+
+// cutome command to validate all success or error messages [Array of messages] or single message
+Cypress.Commands.add('validateSuccessMessages', (messages) => {
+    // Check if it's an array, otherwise treat it as a single message
+    if (Array.isArray(messages)) {
+        messages.forEach((message) => {
+            cy.contains(message).should('be.visible');
```
Comment (Sandy-N) [inline cypress/support/commands.js:59]: fix

**Pull request merged**